### ISSUE TITLE
Improve error message when `pyenv shell system` fails because `python` executable not available on PATH

### ISSUE THREAD

**Issue by RasmusWL** [pyenv/pyenv#1561]
### Description
- [x] Platform information: Ubuntu 18.04
- [x] OS architecture: amd64
- [x] Shell: fish 3.1.0
- [x] pyenv version: pyenv 1.2.17-1-g89786b90
- [x] Python version: N/A
- [x] C Compiler information (e.g. gcc 7.3): N/A
- [x] Please attach verbose build log as gist: https://gist.github.com/RasmusWL/02c26f82008798d0982f46c427d824f0

It seems like pyenv can't handle the special version `system` when there is no `python` executable available on PATH.

```
$ /usr/bin/python3 --version
Python 3.6.9

$ /usr/bin/python2 --version 
/usr/bin/python2: command not found

$ /usr/bin/python --version 
/usr/bin/python: command not found
```

```
$ pyenv versions
* 2.7.17 (set by /home/rasmus/.pyenv/version)
* 3.8.0 (set by /home/rasmus/.pyenv/version)

$ pyenv shell system (debug)
pyenv: system version not found in PATH
```

Gist with debug for [`pyenv versions`](https://gist.github.com/RasmusWL/02c26f82008798d0982f46c427d824f0#file-pyenv-versions) and [`pyenv shell system`](https://gist.github.com/RasmusWL/02c26f82008798d0982f46c427d824f0#file-pyenv-shell-system). The debug log from `pyenv shell system` says that _shell integration not enabled_, but that is not true.

```
$ sudo ln -s /usr/bin/python3 /usr/bin/python

$ pyenv versions
  system
* 2.7.17 (set by /home/rasmus/.pyenv/version)
* 3.8.0 (set by /home/rasmus/.pyenv/version)
```

It is all so obvious now, but man I was confused about this! An error message telling me `python` wasn't available on PATH would probably have saved me a good deal of head-scratching :smile: If there was a command so I could easily disable pyenv, that would also have made debugging a whole lot easier.

I'm also a bit surprised that `pyenv shell system` requires that `python` must exists on PATH, and won't just work if you only have `python3` installed, but I have no clue how the code works, and can't really make a judgement whether that is reasonable or not.

I have used pyenv since I set up my system (I used a minimal installation), and didn't realize that `python` nor `python2` wasn't available without pyenv :sweat_smile: 

P.S. thanks for the great work :heart:

**Comment by truthordata** [comment]
Dude, you just saved me. This definitely needs to be fixed. I have Ubuntu 20.04 with no Python 2, and this stumped me for a while. Nice find! =)

**Comment by native-api** [comment]
Should've been fixed by f77c2d7eab15d3ec04751c4545fabf5d5c56342b

**Comment by RasmusWL** [comment]
I've just installed a new system, that does not have `/usr/bin/python2`, and can confirm that this works now :+1:
